secant/Utils/Zatoshi.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  Zatoshi.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 26.05.2022.
+//
+
+import Foundation
+
+struct Zatoshi {
+    enum Constants {
+        static let oneZecInZatoshi: Int64 = 100_000_000
+        static let maxZecs: Int64 = 21_000_000
+        static let maxZatoshi: Int64 = Constants.oneZecInZatoshi * Constants.maxZecs
+    }
+    
+    static var nsDecimalHandler = NSDecimalNumberHandler(
```

Comment:
How does this work within iOS when it comes to locale changes?

Is this instance frozen with the locale at the time of construction, or does it pick up the current locale during the conversion?

In Java, we construct the formatter with a specific locale and have to create a new instance when the locale changes.

Reply:
Also are there currency specific formatting APIs?

In some countries, I think they use different separators for numbers versus currency.  In Java, we're doing something like this: https://github.com/zcash/secant-android-wallet/blob/94d259d5b0c0175c40dad8b89940bcb4b4e48691/sdk-ext-ui-lib/src/main/java/cash/z/ecc/sdk/ext/ui/model/ZecString.kt#L44

I'm not 100% certain about this, but the fact that there's an explicit `monetaryDecimalSeparator` API in Java indicates it is a consideration.

Reply:
`NSDecimalNumberHandler` is not a formatter. The NSDecimalNumberHandler only defines the strategy for the rounding, fully respecting the locale. 

The NumberFormatter though is used to handle locale formats, floating point separator, currency symbols, etc. In iOS case we are lucky the NumberFormatter is fully automatic and can be used instead of regex you use on Android side. 

So to answer your question, any change in the locale even during runtime is handled by the system, the strategy for rounding stays the same.

Reply:
Awesome, thanks for the clarification!